xilem_web/web_examples/fetch_event_handler/src/main.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use wasm_bindgen::{JsCast, UnwrapThrowExt as _};
+use xilem_web::{
+    document_body,
+    elements::html::*,
+    event_handler::defer,
+    interfaces::{Element, HtmlDivElement, HtmlImageElement},
+    App,
+};
+
+use gloo_net::http::Request;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Cat {
+    pub url: String,
+}
+
+#[derive(Error, Clone, Debug)]
+pub enum CatError {
+    #[error("Please request more than zero cats.")]
+    NonZeroCats,
+}
+
+pub type CatCount = usize;
+
+pub async fn fetch_cats(count: CatCount) -> anyhow::Result<Vec<Cat>> {
+    log::debug!("Fetch {count} cats");
+    if count < 1 {
+        return Err(CatError::NonZeroCats.into());
+    }
+    let url = format!("https://api.thecatapi.com/v1/images/search?limit={count}",);
+    Ok(Request::get(&url)
+        .send()
+        .await?
+        .json::<Vec<Cat>>()
+        .await?
+        .into_iter()
+        .take(count)
+        .collect())
+}
+
+#[derive(Default)]
+struct AppState {
+    cat_count: usize,
+    cats: Vec<Cat>,
+    error: Option<String>,
+}
+
+pub fn event_target_value<T>(event: &T) -> String
```

Comment:
BTW: This is a very common use case, do you think this method should be part of `xilem_web`?

Reply:
Yeah I think we could add more sugar for this. I have also thought about filtering events based on the target, as other events from different targets could reach the event, but I'm not sure, if that may be intended.

Such a function should return an option though in that case to avoid crashing...

Reply:
https://github.com/linebender/xilem/pull/506

Reply:
> Yeah I think we could add more sugar for this. I have also thought about filtering events based on the target, as other events from different targets could reach the event, but I'm not sure, if that may be intended.

In the first step, the unambitious variant would satisfy me :wink: